ecosystem/indexer-grpc/indexer-grpc-integration-test-transaction-generator/src/transaction_generator.rs

```diff
@@ -0,0 +1,273 @@
+                    .parse::<u32>()
+                    .context("Failed to get setup index.")?;
+                setups.push((path, setup_index));
+            } else if file_name.starts_with(TEST_CASE_STEP_PREFIX) {
+                let step_index = file_name
+                    .split('_')
+                    .last()
+                    .context("Failed to get step index.")?
+                    .parse::<u32>()
+                    .context("Failed to get step index.")?;
+                steps.push((path, step_index));
+            }
+        }
+        // Sort the setups and steps.
+        setups.sort_by(|a, b| a.1.cmp(&b.1));
+        steps.sort_by(|a, b| a.1.cmp(&b.1));
+        tracing::info!("Setups: {:?}", setups);
+        tracing::info!("Steps: {:?}", steps);
+
+        let move_executor = self
+            .move_executor
+            .as_ref()
+            .expect("Move executor is not initialized.");
+
+        // Execute the setups.
+        for (setup_path, _) in setups {
+            let _ = move_executor
+                .execute(&setup_path)
+                .await
+                .context(format!("Failed to execute setups {:?}", setup_path))?;
+        }
+        let mut transactions = Vec::new();
+
+        // Execute the steps.
+        for (step_path, _) in steps {
+            let transaction = move_executor
+                .execute(&step_path)
+                .await
+                .context(format!("Failed to execute steps {:?}", step_path))?
+                .expect("Transaction is not found.");
+            transactions.push(transaction);
+        }
+
+        // Capture the transactions.
+        let output_folder = self
+            .output_test_cases_folder
+            .join(
+                test_case_path
+                    .file_name()
+                    .context("Failed to get test name")?,
+            )
+            .join(&self.node_version);
+        capture_transactions(transactions, output_folder)
+            .await
+            .context(format!(
+                "Failed to capture transactions{:?}",
+                test_case_path
+            ))?;
+
+        Ok(())
+    }
+}
+
+async fn start_localnode(path: PathBuf, node_config: Option<PathBuf>) -> anyhow::Result<Child> {
+    // Start the local node.
+    let mut node_process_cmd = tokio::process::Command::new(path);
+    // TODO: supports node config.
+    node_process_cmd
+        .arg("node")
+        .arg("run-local-testnet")
+        .arg("--force-restart")
+        .arg("--assume-yes");
+
+    // Feed the node config if provided.
+    if let Some(path) = node_config {
+        node_process_cmd.arg("--config").arg(path);
+    }
+
+    let node_process = node_process_cmd
+        // TODO: fix this with child.kill().
+        .kill_on_drop(true)
+        .spawn()
+        .context("Failed to start local node.")?;
+    for _ in 0..NODE_HEALTH_CHECK_COUNT {
+        // Curl faucet to make sure the node is up.
```

Comment:
The localnet actually has a healthcheck endpoint that you can use. Instead of checking the node and faucet separately, just use that. It runs at port 8070, it will return 200 when everything is ready.